core/embed/rust/src/translations/flash.rs

```diff
@@ -1,28 +1,40 @@
+use spin::{RwLock, RwLockReadGuard};
+
 use crate::{error::Error, trezorhal::translations};
 
 use super::blob::Translations;
 
-static mut TRANSLATIONS_ON_FLASH: Option<Translations> = None;
+static TRANSLATIONS_ON_FLASH: RwLock<Option<Translations>> = RwLock::new(None);
```

Comment:
Why use a spinlock if we only have one core? Wouldn't a `RefCell` work as well?

Reply:
I'm not sure. maybe it would?
how would you implement get() with a refcell?

Reply:
Something like this?
```rs
    TRANSLATIONS_ON_FLASH
        .try_borrow()
        .map_err(|| value_error!("Translations are in use."))
```
Maybe I'm missing something but from far enough away `RefCell` looks like a single-thread `RwLock` that panics instead of spinning forever.

Reply:
Ah, the issue is, stupidly, that `RefCell` is not `Sync`. The compiler is straight up recommending RwLock instead.
I'm gonna replace all locks with `unwrap!(try_xx)` to make it error out instead of spin

Reply:
a6bec64b71c96129382a08e8a464405ecf9bab3a

Reply:
because statics are mandatory sync (and also maybe send)